LFS171x/indy-material/nodejs/AgentDemoScript.md

```diff
@@ -0,0 +1,319 @@
+
+> **NOTE:** From time to time in running the demo you must refresh the Browser (Crl-R or Cmd-R) to update the screen data. This is needed when an action by one Identity (e.g. Alice) affects the data of another (e.g. Faber).
+
+
+### Step 2: Setting up Faber
+
+Before we get started, Faber has to do some one time setup that we'll do first.  Go to the Faber browser tab (localhost:3002) and click **Issuing**. We're going to do two things here:
+
+
+
+*   First, we'll post to the Hyperledger Indy Ledger a transcript _Schema_. This is a public definition of what fields (claims) are in Transcript Credentials that Faber will issue to its graduates. Since it is public, the Schema could be published by anyone (such as the "Association of Colleges and Universities") and can be used by anyone (such as other institutions like "Port Chester University").
+*   Second, we'll post to the Hyperledger Indy Ledger a "Credential Definition," an Indy component that links a particular Credential Issuer (Faber in this case), with a Schema (the Transcript Schema), an optional Revocation Registry (we're not going to use that) and a series of Keys used for encryption and signing of the Claims in the Credentials.
+
+Together, that public data is what allows Faber to issue a Transcript Credential to Alice that she can later prove to Acme.
+
+Back to our story.  Faber has just set up a Hyperledger Indy Agent that can issue credentials. John Academic, Faber's most awesome transcript keeper, needs to get things working. He will have to write a Schema and Credential Definition to the Indy Ledger to do this. It is a bit complicated but nothing John Academic can't handle!
+
+
+
+1.  If you haven't already done so, go to the Faber agent tab and click **Issuing**.
+
+|![CreateSchema](images/CreateSchema.png "Publish schema to the Indy ledger")|
+|:--:|
+|*Licensed under [CC By 4.0](https://creativecommons.org/licenses/by/4.0/)*|
+
+1.  Give the schema a name, a version and a list of attributes. For example, _Transcript 1.3 with name, degree, status, year, average and social security number._
+2.  Click **Submit**.
+
+This writes the Schema to the Ledger so Faber (and others) can use it. 
+
+
+
+1.  In the Credential Definition window, select a **Schema** (Transcript 1.3), tag it and click **Submit** and wait for the action to complete (which takes a bit…).
+
+|![CreateCredDef](images/CreateCredDef.png "Create Faber Credential Definition")|
+|:--:|
+|*Licensed under [CC By 4.0](https://creativecommons.org/licenses/by/4.0/)*|
+
+John tagged his credential as MyTranscript. John probably could have come up with a better name, but hey, he's not _that_ smart.

```

Comment:
I attempted to provide a better name (FaberTranscript). When I did that, I could not get the information needed for Acme to confirm the transcript information. Might be worth a note (or the bug needs to be fixed if someone chooses something other than `MyTranscript`.

Reply:
Clarified that it should not be changed.